assets/app/view/route_selector.rb

```diff
@@ -28,36 +35,75 @@ def render
         style = {
           border: "solid #{selected ? '3px' : '1px'} rgba(0,0,0,0.2)",
           display: 'inline-block',
-          cursor: selected ? 'none' : 'pointer',
+          cursor: selected ? 'default' : 'pointer',
           margin: '0.5rem 0.5rem 0.5rem 0',
           padding: '0.5rem',
         }
 
-        h(:div, { style: style, on: { click: onclick } }, "Train: #{train.name}")
+        route = @routes.find { |t| t.train == train }
+        children = []
+        if route && !route.hexes.empty?
+          revenue, invalid = begin
+                               [@game.format_currency(route.revenue), nil]
+                             rescue Engine::GameError => e
```

Comment:
why would there be a game error

Reply:
route.revenue throws an exception if it can't run (to long a route etc)